src/browser/HostInstaller.cpp

```diff
@@ -0,0 +1,189 @@
+    const QString HostInstaller::TARGET_DIR_FIREFOX = "/.mozilla/native-messaging-hosts";
+    const QString HostInstaller::TARGET_DIR_VIVALDI = "/.config/vivaldi/NativeMessagingHosts";
+#elif defined(Q_OS_WIN)
+    const QString HostInstaller::TARGET_DIR_CHROME = "HKEY_CURRENT_USER\\Software\\Google\\Chrome\\NativeMessagingHosts\\" + HostInstaller::HOST_NAME;
+    const QString HostInstaller::TARGET_DIR_CHROMIUM = "HKEY_CURRENT_USER\\Software\\Chromium\\NativeMessagingHosts\\" + HostInstaller::HOST_NAME;
+    const QString HostInstaller::TARGET_DIR_FIREFOX = "HKEY_CURRENT_USER\\Software\\Mozilla\\NativeMessagingHosts\\" + HostInstaller::HOST_NAME;
+    const QString HostInstaller::TARGET_DIR_VIVALDI = "HKEY_CURRENT_USER\\Software\\Vivaldi\\NativeMessagingHosts\\" + HostInstaller::HOST_NAME;
+#endif
+
+HostInstaller::HostInstaller()
+{
+
+}
+
+bool HostInstaller::checkIfInstalled(const supportedBrowsers browser)
+{
+    QString fileName = getPath(browser);
+#ifdef Q_OS_WIN
+    QSettings settings(getTargetPath(browser), QSettings::NativeFormat);
+    if (registryEntryFound(settings)) {
+        return true;
+    }
+    return false;
+#else
+    if (QFile::exists(fileName)) {
+        return true;
+    }
+    return false;
+#endif
+}
+
+void HostInstaller::installBrowser(const supportedBrowsers browser, const bool enabled)
+{
+    if (enabled && !checkIfInstalled(browser)) {
+        QJsonObject script = constructFile(browser);
+#ifdef Q_OS_WIN
+        // Create a registry key
+        QSettings settings(getTargetPath(browser), QSettings::NativeFormat);
+        if (!registryEntryFound(settings)) {
+            settings.setValue("Default", getPath(browser));
+        }
+#endif
+        // Install the .json file
+        if (!saveFile(browser, script)) {
+
+        }
+    } else if (!enabled && checkIfInstalled(browser)) {
+        // Uninstall the .json file
+        QString fileName = getPath(browser);
+        QFile::remove(fileName);
+#ifdef Q_OS_WIN
+        // Remove the registry entry
+        QSettings settings(getTargetPath(browser), QSettings::NativeFormat);
+        if (registryEntryFound(settings)) {
+            settings.remove("Default");
+        }
+#endif
+    }
+}
+
+QString HostInstaller::getTargetPath(const supportedBrowsers browser)
+{
+    switch (browser) {
+        case supportedBrowsers::CHROME:     return HostInstaller::TARGET_DIR_CHROME;
+        case supportedBrowsers::CHROMIUM:   return HostInstaller::TARGET_DIR_CHROMIUM;
+        case supportedBrowsers::FIREFOX:    return HostInstaller::TARGET_DIR_FIREFOX;
+        case supportedBrowsers::VIVALDI:    return HostInstaller::TARGET_DIR_VIVALDI;
+        default: return "";
+    }
+}
+
+QString HostInstaller::getPath(const supportedBrowsers browser)
+{
+#ifdef Q_OS_WIN
+    QString winPath = (browser == supportedBrowsers::FIREFOX)
+        ? QString("%1/%2_firefox.json").arg(QCoreApplication::applicationDirPath(), HostInstaller::HOST_NAME)
```

Comment:
As KeePass is normally installed in Program Files, you need administrator privileges to write. Consider a better location or request UAC.

Reply:
I'll move the script file to user's temporary directory. Thanks for pointing this out.

Reply:
Just a note: I'm not sure if what you tried to say, but temporary directory can be cleared by cleanup utilities. AppData is fine.

Reply:
I ended up using the same directory where keepassxc.ini exists, so under AppData. Also the .json files are named for each browser separately. Pushing in the fixes today.